galsim/utilities.py

```diff
@@ -418,3 +418,133 @@ def _gammafn(x):
+            raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+
+    if len(im_list)<2:
+        raise TypeError("'im_list' needs to have at least two instances of galsim.Image")
+
+    if (n1*n2 != len(im_list)):
+        raise ValueError("'N' is incompatible with the number of images in 'im_list'")
+
+    if len(im_list)!=len(offsets):
+        raise ValueError("'im_list' and 'offsets' must be lists of same length")
+    for offset in offsets:
+        if not isinstance(offset,galsim.PositionD):
+            raise TypeError("'offsets' must be a list of galsim.PositionD instances")
+
+    if isinstance(im_list[0],galsim.Image):
+        y_size, x_size = im_list[0].array.shape
+        scale = im_list[0].scale
+    else:
+        raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+    for im in im_list[1:]:
+        if not isinstance(im,galsim.Image):
+            raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+        if im.array.shape != (y_size,x_size):
+            raise ValueError("All galsim.Image instances in 'im_list' must be of the same size")
+ 
+        if im.scale != scale:
+            raise ValueError(
+                "All galsim.Image instances in 'im_list' must have the same pixel scale")
+
+    img_array = np.zeros((n2*y_size,n1*x_size))
+    # The tricky part - going from (x,y) Image coordinates to array indices
+    # DX[i'] = -(i+0.5)/n+0.5 = -i/n + 0.5*(n-1)/n
+    #    i  = -n DX[i'] + 0.5*(n-1)
+    for k in xrange(len(offsets)):
+        dx, dy = offsets[k].x, offsets[k].y
+
+        i = int(round((n1-1)*0.5-n1*dx))
+        j = int(round((n2-1)*0.5-n2*dy))
+
+        if catch_offset_errors is True:
+            err_i = (n1-1)*0.5-n1*dx - round((n1-1)*0.5-n1*dx)
+            err_j = (n2-1)*0.5-n2*dy - round((n2-1)*0.5-n2*dy)
+            tol = 1.e-6
+            if abs(err_i)>tol or abs(err_j)>tol:
+                raise ValueError("'offsets' must be a list of galsim.PositionD instances with x "
+                            +"values spaced by 1/{0} and y values by 1/{1} around 0 for N = ".format(n1,n2)+str(N))
+
+            if i<0 or j<0 or i>=x_size or j>=y_size:
+                raise ValueError("'offsets' must be a list of galsim.PositionD instances with x "
+                            +"values spaced by 1/{0} and y values by 1/{1} around 0 for N = ".format(n1,n2)+str(N))
+
+        img_array[j::n2,i::n1] = im_list[k].array[:,:]
+
+    if add_flux is True:
+        img = galsim.Image(img_array)
+    else:
+        img = galsim.Image((1.0/(len(im_list)))*img_array)
+
+    if (n1==n2):
+        if scale is not None:
+            img.scale = im_list[0].scale*(1./n1)
+    elif suppress_warnings is False:
+        import warnings
+        warnings.warn("Interleaved image could not be assigned a pixel scale automatically")
```

Comment:
Au contraire.  If the originals did have PixelScale WCS's, then the appropriate final WCS is
```
img.wcs = galsim.Jacobian(scale/n1, 0., 0., scale/n2)
```
Also, I'd put the check for `if wcs is not None` at the start, since the warning doesn't make much sense in that case, as the scale couldn't be assigned automatically even if n1 == n2 then.